runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -450,7 +450,7 @@ MM_VLHGCAccessBarrier::copyStringCritical(J9VMThread *vmThread, GC_ArrayObjectMo
 	} else {
 		if (isCompressed) {
 			for (jint i = 0; i < length; i++) {
-				*data[i] = (jchar)(U_8)J9JAVAARRAYOFBYTE_LOAD(vmThread, (j9object_t)valueObject, i);
+				(*data)[i] = (jchar)(U_8)J9JAVAARRAYOFBYTE_LOAD(vmThread, (j9object_t)valueObject, i);
```

Comment:
We should using another variable and take `*data` out of the loop.

Reply:
Added another variable here.